### PR TITLE
Check Mustache templates in AI question generation

### PR DESCRIPTION
**Pull request by jlevine18**
This PR updates the question validation used in AI question generation to check Mustache templates for validity. We observed cases where it would try to use logic in Mustache templates, like `{{ not params.foo }}` or `{{ params.too == true }}`, which would fail to do anything useful at runtime.

### CODE REVIEW

File: apps/prairielearn/src/ee/lib/validateHTML.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ type ChildNode = parse5.DefaultTreeAdapterMap['childNode'];
 
 const mustacheTemplateRegex = /^\{\{.*\}\}$/;
 const mustacheTemplateExtractorRegex = /\{\{((?:[^}]|\}[^}])*)\}\}/g;
+const mustacheTemplateChecker = /^\s*(((#|\/|\^|>|!|\$|>\*)?\s*(\w|\d|\.|_)*)|!.*)\s*$/;
```
Comment (mwest1066) [inline apps/prairielearn/src/ee/lib/validateHTML.ts:8]: Rather than using regexes here, can we use a similar approach to #11962?

File: apps/prairielearn/src/ee/lib/validateHTML.ts [hunk 2]
```diff
@@ -645,5 +646,13 @@ export function validateHTML(file: string, optimistic: boolean, usesServerPy: bo
...
   }
 
+  for (const template of templates) {
+    if (!mustacheTemplateChecker.test(template)) {
+      errors.push(
+        `Template of ${template} must be in mustache format. Please adjust so that any logic done in the template is instead done in server.py`,
+      );
+    }
+  }
+
```
Comment (mwest1066) [inline apps/prairielearn/src/ee/lib/validateHTML.ts:656]: Can we include some tests, both positive and negative?

File: apps/prairielearn/src/ee/lib/validateHTML.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/ee/lib/validateHTML.ts:8]: @mwest1066 ha, that's precisely why I opened that PR first, I'm going to rebuild this one on top of that. The implementation should get much simpler.

File: apps/prairielearn/src/ee/lib/validateHTML.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/ee/lib/validateHTML.ts:656]: I started trying to do this when I realized that there were some issues both with the existing and new implementation, which is actually where https://github.com/PrairieLearn/PrairieLearn/pull/11962 came from. I'll add tests once this PR is rebased on that.

File: apps/prairielearn/src/ee/lib/validateHTML.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/ee/lib/validateHTML.ts]: I added tests in f99e574.

File: apps/prairielearn/src/ee/lib/validateHTML.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/ee/lib/validateHTML.ts:691]: I added tests in f99e574.

Comment (mwest1066) [reply apps/prairielearn/src/ee/lib/validateHTML.ts:691]: Thanks!

File: apps/prairielearn/src/ee/lib/validateHTML.ts (hunk 1 above)
Comment (mwest1066) [reply apps/prairielearn/src/ee/lib/validateHTML.ts]: Very nice!

**Review by mwest1066 — APPROVED** [review]
